### PR TITLE
ROFF2 bug fixes

### PR DESCRIPTION
**Pull request by Archangel35757**
Fixes for issues #1002 and #1003.  Additional ROFF2 Notetrack enhancements-- added support for new Notetrack types:  USE, loop

Supported notetrack types:    effect, sound, USE, loop

General notetrack format:        \<type\> \<argument\> [additionalArguments]
Note: <> denote required argument, [ ] denote optional argument, | denotes argument choices

Examples:
effect notetrack format:        effect <relative_effectfilepath.efx> [originOffset] [rotationOffset]
                    notetrack = "effect effects/explosion1.efx 0+0+64 0-0-1";
'effect' notes:
        (1) the '+' and '-' are delimiters and not positive/negative signs. For example, negative origin offset would be: -10+-20+-10
        (2) angles are expected to be from 0 to 360, i.e., no negative angles.
        (3) optional additional argument for rotationOffset requires the originOffset preceding it.

sound notetrack format:        sound <relative_soundfilepath.ext>
                    notetrack =    "sound sound/vehicles/tie/flyby2.mp3";
'sound' notes:
         (1) supported sound file formats are: .mp3, .wav

USE notetrack format:            USE <relative_scriptfilepath_noExt>
                    notetrack = "USE shuttlemap/shuttletakeoff";

loop notetrack format:        loop rof < absolute | relative >
                                          loop sfx < relative_soundfilepath.ext | kill >
                    notetrack = "loop rof absolute";
                    notetrack = "loop rof relative";
                    notetrack = "loop sfx sound/vehicles/tie/loop.wav";
                    notetrack = "loop sfx kill";
'loop rof' notes:  
        (1) absolute ==> reset rof to original delta position/rotation world location before looping.
        (2) relative ==> reset rof to original delta position/rotation at current location before looping.
'loop sfx' notes:
        (1) adds a sound to be looped which gets assigned to the entitystate's loopSound parameter.
        (2) addlArg 'kill' -- kills the looping sound by setting entitystate's loopSound equal to zero.

****Example Test Level****
[shuttlemap.zip](https://github.com/JACoders/OpenJK/files/3175127/shuttlemap.zip)

### CODE REVIEW

File: code/game/g_roff.cpp [hunk 1]
```diff
@@ -238,34 +277,176 @@ static void G_RoffNotetrackCallback( gentity_t *cent, const char *notetrack)
 			useOrigin[1] += up[1]*parsedOffset[2];
 			useOrigin[2] += up[2]*parsedOffset[2];
 
+#if !NDEBUG
```
Comment (Razish) [inline code/game/g_roff.cpp]: Debug prints no longer needed? Suggest removing them all

**Comment by Archangel35757** [comment]
@Razish ...would you not want to retain those for debug builds?

File: code/game/g_roff.cpp (hunk 1 above)
Comment (Archangel35757) [reply code/game/g_roff.cpp]: @Razish ...would you not want to retain those for debug builds?

Comment (Archangel35757) [reply code/game/g_roff.cpp]: @Razish -- ok... the code has been updated to remove the debug-build prints and replaced with an if statement using g_developer->integer.  Should be good to go now.

**Comment by Archangel35757** [comment]
Looks like I need to fix a formatting issue at line 308... spaces versus tabs or something.  The else if USE block needs tabbing.

File: code/game/g_roff.cpp [hunk 2]
```diff
@@ -238,34 +279,187 @@ static void G_RoffNotetrackCallback( gentity_t *cent, const char *notetrack)
...
+		//run the IBI script
+		Quake3Game()->RunScript(ent, argument);
+	}
+	else if (strcmp(type, "loop") == 0)
+	{
+		if (strcmp(argument, "rof") == 0)
+		{
+			if (strcmp(addlArg, "absolute") == 0)
+			{
+				VectorSubtract(ent->pos1, ent->pos1, ent->pos1);
```
Comment (xycaleth) [inline code/game/g_roff.cpp]: No need to change, but this code is very strange!

It's the same as:
```c++
VectorClear(ent->pos1);
VectorClear(ent->pos2);
VectorClear(ent->s.pos.trBase);
VectorClear(ent->s.apos.trBase);

VectorCopy(ent->currentOrigin, ent->s.origin2);
VectorCopy(ent->currentAngles, ent->s.angles2);
```

**Comment by xycaleth** [comment]
Thanks for the contribution, @Archangel35757! 😁

**Pull request merged**